File: Adapters/Database/PostgresMerchantRepository.cs
using MerchantOnboarding.Domain;
using MerchantOnboarding.Domain.Repository;
using MerchantOnboarding.Infrastructure;
using Npgsql;

namespace MerchantOnboarding.Adapters.Database;

public class PostgresMerchantRepository:MerchantRepository
{

    private readonly MerchantDbContext _context;
    
    public PostgresMerchantRepository(MerchantDbContext context)
    {
        this._context = context;
    }
    
    public void Save(Merchant merchant)
    {
        this._context.Merchants.Add(merchant);
        this._context.SaveChanges();
    }

    public Merchant Get(Guid id)
    {
        return this._context.Merchants.Find(id);
    }

    public List<Merchant> Merchants()
    {
        return this._context.Merchants.ToList();
    }
    
}

File: src/main/java/com/example/merchantonboarding/infrastructure/repository/PostgresMerchantRepository.java
package com.example.merchantonboarding.infrastructure.repository;

import com.example.merchantonboarding.domain.Merchant;
import com.example.merchantonboarding.domain.repository.MerchantRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.UUID;

@Repository
public class PostgresMerchantRepository implements MerchantRepository {
    
    private final JpaMerchantRepository jpaMerchantRepository;
    
    @Autowired
    public PostgresMerchantRepository(JpaMerchantRepository jpaMerchantRepository) {
        this.jpaMerchantRepository = jpaMerchantRepository;
    }
    
    @Override
    public void save(Merchant merchant) {
        jpaMerchantRepository.save(merchant);
    }
    
    @Override
    public Merchant findById(UUID id) {
        return jpaMerchantRepository.findById(id).orElse(null);
    }
    
    @Override
    public List<Merchant> findAll() {
        return jpaMerchantRepository.findAll();
    }
}